artemis-commons/src/main/java/org/apache/activemq/artemis/core/server/ActiveMQScheduledComponent.java

```diff
@@ -60,41 +58,58 @@
     * @param checkPeriod              the delay between the termination of one execution and the start of the next
     * @param timeUnit                 the time unit of the {@code initialDelay} and {@code checkPeriod} parameters
     * @param onDemand                 if {@code true} the task won't be scheduled on {@link #start()}, {@code false} otherwise
+    * @param overBooking              if {@code true} {@link #delay()} executions and period ones are allowed to accumulate, {@code false} otherwise. Default is {@code false}.
     */
    public ActiveMQScheduledComponent(ScheduledExecutorService scheduledExecutorService,
                                      Executor executor,
                                      long initialDelay,
                                      long checkPeriod,
                                      TimeUnit timeUnit,
-                                     boolean onDemand) {
+                                     boolean onDemand,
+                                     boolean overBooking) {
```

Comment:
> I would remove the overbooking, eliminate the possibility of null for the atomic, and simplify the rest of the code accordingly. It will still be plenty complicated :)

+100 agree: I was thinking that was needed for the JDBC lease lock, but effectively isn't needed, I'm removing it